src/sentry/uptime/consumers/results_consumer.py

```diff
@@ -77,14 +77,12 @@ def _get_snuba_uptime_checks_producer() -> KafkaProducer:
 _snuba_uptime_checks_producer = SingletonProducer(_get_snuba_uptime_checks_producer)
 
 
-def build_last_update_key(project_subscription: ProjectUptimeSubscription) -> str:
-    return f"project-sub-last-update:{project_subscription.id}"
+def build_last_update_key(detector: Detector) -> str:
+    return f"project-sub-last-update:detector:{detector.id}"
```

Comment:
When we merge this it's possible we could duplicate checks. I don't think this will be a problem though and probably isn't worth trying to mitigate.

Reply:
Added the `detector:` prefix to ensure we don't lookup a mis-matched id.

Reply:
Yeah, seems ok to me.